jung-samples/src/main/java/edu/uci/ics/jung/samples/ShortestPathDemo.java

```diff
@@ -75,10 +76,16 @@ public void paint(Graphics g) {
                 EndpointPair<String> endpoints = mGraph.incidentNodes(e);
                 String v1 = endpoints.nodeU();
                 String v2 = endpoints.nodeV();
-                Point2D p1 = layoutModel.apply(v1);
-                Point2D p2 = layoutModel.apply(v2);
-                p1 = vv.getRenderContext().getMultiLayerTransformer().transform(Layer.LAYOUT, p1);
-                p2 = vv.getRenderContext().getMultiLayerTransformer().transform(Layer.LAYOUT, p2);
+                Point p1 = layoutModel.apply(v1);
+                Point p2 = layoutModel.apply(v2);
+                Point2D p2d1 =
+                    vv.getRenderContext()
+                        .getMultiLayerTransformer()
+                        .transform(Layer.LAYOUT, new Point2D.Double(p1.x, p1.y));
```

Comment:
I like this because my goal in general is to use the new Point class all over the place in jung-algorithms, but use it as little as possible in the awt domain, preferring to receive the (x,y) instead of the Point and trying to never create new Points on the awt side (outside of places in jung-samples). It will probably take a few passes thru to code to clean up everything.